scriptmodules/libretrocores/lr-scummvm.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="lr-scummvm"
+rp_module_desc="ScummVM port for libretro"
+rp_module_help="Copy your ScummVM games to $romdir/scummvm\n\nThe name of your game directories must be suffixed with '.svm' for direct launch in EmulationStation."
+rp_module_licence="GPL2 https://raw.githubusercontent.com/libretro/scummvm/master/COPYING"
+rp_module_section="exp"
+
+function sources_lr-scummvm() {
+    gitPullOrClone "$md_build" https://github.com/libretro/scummvm.git
+    # fix random freeze-on-quit (remove when patched upstream)
+    applyPatch "$md_data/01_fix_quit.diff"
+}
+
+function build_lr-scummvm() {
+    make clean -C backends/platform/libretro/build
+
+    local params=(HAVE_MT32EMU=1)
+    local platform
+    isPlatform "arm" && platform+="armv"
+    isPlatform "neon" && platform+="neon"
+    [[ -n "$platform" ]] && params+=(platform="$platform")
+
+    CXXFLAGS="$CXXFLAGS -DHAVE_POSIX_MEMALIGN=1" LDFLAGS="$LDFLAGS -shared -Wl,--no-undefined" \
+        make "${params[@]}" -C backends/platform/libretro/build
+    md_ret_require="$md_build/backends/platform/libretro/build/scummvm_libretro.so"
+}
+
+function install_lr-scummvm() {
+    md_ret_files=(
+        "backends/platform/libretro/build/scummvm_libretro.so"
+        "COPYING"
+    )
+}
+
+function configure_lr-scummvm() {
+    addEmulator 0 "romdir-$md_id" "scummvm" "$md_inst/retroarch-romdir.sh %ROM%"
```

Comment:
why romdir-$md_id ? $md_id should suffice.

Reply:
No, as noted in https://github.com/RetroPie/RetroPie-Setup/pull/2537#issuecomment-443906376, the `addEmulator` function has the shortcoming that, when $md_id starts with `lr-`, then it overwrites the cmd passed with a static retroarch invocation. The affecting conditional is this:
```sh
    # automatically add parameters for libretro modules
    if [[ "$id" == lr-* && "$cmd" != "$emudir/retroarch/bin/retroarch"* ]]; then
        cmd="$emudir/retroarch/bin/retroarch -L $cmd --config $md_conf_root/$system/retroarch.cfg %ROM%"
    fi
```

I noticed that no scriptmodule benefits from the second part of the conditional, so maybe the conditional can be improved to check for a passed libretro core (`*.so`) instead of `retroarch`. If you agree with that, I actually would prefer that solution, I don't like the id name :)

Note that for `runcommand` to recognise a retroarch core launching, the name fo the cmd needs to have `retroarch` in its name.

Reply:
I'll come up with something.